src/Uno.UI/Microsoft/UI/Xaml/Controls/TabView/TabView.cs

```diff
@@ -0,0 +1,1322 @@
+			}
+
+			if (m_scrollDecreaseButton != null)
+			{
+				m_scrollDecreaseButton.Click -= OnScrollDecreaseClick;
+			}
+
+			if (m_scrollIncreaseButton != null)
+			{
+				m_scrollIncreaseButton.Click -= OnScrollIncreaseClick;
+			}
+
+			m_tabContentPresenter = null;
+			m_rightContentPresenter = null;
+			m_leftContentColumn = null;
+			m_tabColumn = null;
+			m_addButtonColumn = null;
+			m_rightContentColumn = null;
+			m_tabContainerGrid = null;
+			m_shadowReceiver = null;
+			m_listView = null;
+			m_addButton = null;
+			m_itemsPresenter = null;
+			m_scrollViewer = null;
+			m_scrollDecreaseButton = null;
+			m_scrollIncreaseButton = null;
+		}
+
+		private void OnTabWidthModePropertyChanged(DependencyPropertyChangedEventArgs args)
+		{
+			UpdateTabWidths();
+
+			foreach (var item in TabItems)
+			{
+				// Switch the visual states of all tab items to the correct TabViewWidthMode
+				TabViewItem GetTabViewItem(object item)
+				{
+					var tabViewItem = item as TabViewItem;
+					if (tabViewItem != null)
+					{
+						return tabViewItem;
+					}
+					return ContainerFromItem(item) as TabViewItem;
+				}
+				var tvi = GetTabViewItem(item);
+
+				if (tvi != null)
+				{
+					tvi.OnTabViewWidthModeChanged(TabWidthMode);
+				}
+			}
+		}
+
+		private void OnCloseButtonOverlayModePropertyChanged(DependencyPropertyChangedEventArgs args)
+		{
+			// Switch the visual states of all tab items to to the correct closebutton overlay mode
+			foreach (var item in TabItems)
+			{
+				TabViewItem GetTabViewItem(object item)
+				{
+					var tabViewItem = item as TabViewItem;
+					if (tabViewItem != null)
+					{
+						return tabViewItem;
+					}
+					return ContainerFromItem(item) as TabViewItem;
+				}
+				var tvi = GetTabViewItem(item);
+
+				if (tvi != null)
+				{
+					tvi.OnCloseButtonOverlayModeChanged(CloseButtonOverlayMode);
+				}
+			}
+		}
+
+		private void OnAddButtonClick(object sender, RoutedEventArgs args)
+		{
+			AddTabButtonClick?.Invoke(this, args);
+		}
+
+		protected override AutomationPeer OnCreateAutomationPeer()
+		{
+			return new TabViewAutomationPeer(this);
+		}
+
+		private void OnLoaded(object sender, RoutedEventArgs args)
+		{
+			UpdateTabContent();
+		}
+
+		void OnListViewLoaded(object sender, RoutedEventArgs args)
+		{
+			var listView = m_listView;
+			if (listView != null)
+			{
+				// Now that ListView exists, we can start using its Items collection.
+				var lvItems = listView.Items;
+				if (lvItems != null)
+				{
+					// TODO: Uno specific - Items collection is not yet in sync with ItemsSource
+					// this is a workaround
+
+					if (listView.ItemsSource == null)
+					{
+						var observableTabItems = new ObservableVector<object>();
+						foreach(var item in TabItems)
+						{
+							observableTabItems.Add(item);
+						}
+						listView.ItemsSource = observableTabItems;
+						TabItems = observableTabItems;
+					}
+					
+					//if (listView.ItemsSource == null)
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6063676)